python/smqtk/detect_and_describe/__init__.py

```diff
@@ -0,0 +1,64 @@
+import abc
+import os
+
+from smqtk.data_rep import DataElement
+from smqtk.utils.plugin import get_plugins
+
+class DetectAndDescribe (object):
+	def __init__(self, descriptor_element_factory, data_element_type):
+		"""
+		Base abstract initializer for a DetectAndDescribeObject. 
+		:param descriptor_element_factory: A DescriptorElementFactory that will create the appropriate DescriptorElement 
+			to be returned from the implementation of this method.
+		:type descriptor_element_factory: DescriptorElementFactory:
+		"""
+		pass
+
+	@abc.abstractmethod
+	def detect_and_describe(self, data):
+		"""
+		This method is to be implemented in classes that implement DetectAndDescribe. It should do what the name
+		implies, given input data -- 
+			1) sample the points (i.e., detect)
+			2) take the sampled points and compute the implemented descriptor (i.e., describe)
+
+		Of course, an implementation of DetectAndDescribe does not have to have seperate detect and describe
+		stages, as would be the case for some implementataions such as colorDescriptor.
+
+		:param data: DataSet or DataElement (which will be converted to a DataSet) on which we will perform
+			feature detection and description.
+		:type data: DataSet (or DataElement)
```

Comment:
This should probably expect an abstract iterable of `DataElement` instances. This will cover the cases of one or more `DataElement` instances being passed (e.g. `:type data: collections.Iterable[DataElement]`).